### ISSUE TITLE
The "trap disarm" skill works every time

### ISSUE THREAD

**Issue by Chance4us** [diasurgical/devilutionX#1747]
I can't remember that in vanilla rogue was able to successfully disarm the traps every time.

**Comment by qndel** [comment]
```cpp
void TryDisarm(int pnum, int i)
{
    int j, oi, oti, trapdisper;
    bool checkflag;

    if (pnum == myplr)
        NewCursor(CURSOR_HAND);
    if (object[i]._oTrapFlag) {
        trapdisper = 2 * plr[pnum]._pDexterity - 5 * currlevel;
        if (GenerateRnd(100) <= trapdisper) {
```

After getting 110 dex you have 100% chance to disarm any trap, please do some research and/or provide info when opening issues

**Comment by julealgon** [comment]
@Chance4us in vanilla, there was a bug that caused the rogue to not always disarm traps in some situations. That bug was fixed in DevilutionX and we now always respect the correct formula for disarming.

**Comment by qndel** [comment]
That bug was fixed in 1.09 but was still present in hellfire @julealgon
